crates/forge/tests/integration/mock_call.rs

```diff
@@ -207,3 +207,376 @@ fn mock_calls() {
+
+            let mock_ret_data = 421;
+            let expected_calldata = MockCalldata::Values([].span());
+            mock_call_when(contract_address, selector!("get_thing"), expected_calldata, mock_ret_data, 1);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 420);
+        }
+
+        #[test]
+        fn mock_call_when_twice_specific() {
+            let calldata = array![420];
+
+            let contract = declare("MockChecker").unwrap().contract_class();
+            let (contract_address, _) = contract.deploy(@calldata).unwrap();
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+
+            let mock_ret_data = 421;
+            let expected_calldata = MockCalldata::Values([].span());
+            mock_call_when(contract_address, selector!("get_thing"), expected_calldata, mock_ret_data, 2);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 420);
+        }
+
+        #[test]
+        fn mock_call_when_one_any() {
+            let calldata = array![420];
+
+            let contract = declare("MockChecker").unwrap().contract_class();
+            let (contract_address, _) = contract.deploy(@calldata).unwrap();
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+
+            let mock_ret_data = 421;
+            mock_call_when(contract_address, selector!("get_thing"), MockCalldata::Any, mock_ret_data, 1);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 420);
+        }
+
+        #[test]
+        fn mock_call_when_twice_any() {
+            let calldata = array![420];
+
+            let contract = declare("MockChecker").unwrap().contract_class();
+            let (contract_address, _) = contract.deploy(@calldata).unwrap();
+
+            let dispatcher = IMockCheckerDispatcher { contract_address };
+
+            let mock_ret_data = 421;
+            mock_call_when(contract_address, selector!("get_thing"), MockCalldata::Any, mock_ret_data, 2);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 421);
+
+            let thing = dispatcher.get_thing();
+            assert_eq!(thing, 420);
+        }
+
+    "#
+        ),
+        Contract::from_code_path(
+            "MockChecker".to_string(),
+            Path::new("tests/data/contracts/mock_checker.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+    assert_passed(&result);
+}
+
+#[test]
+fn mock_calls_when_mixed() {
```

Comment:
Can we add some cases where the same tests uses both `mock_call` and `mock_call_when`?

E.g. in case where you do `start_mock_call_when(MockCalldata::Any, ...)` and then `stop_mock_call(...)` and similar?

Reply:
How about this @ptisserand ?